src/controllers/hid/hidioreport.cpp

```diff
@@ -0,0 +1,117 @@
+#include "controllers/hid/hidioreport.h"
+
+#include <hidapi.h>
+
+#include "controllers/defs_controllers.h"
+#include "controllers/hid/legacyhidcontrollermappingfilehandler.h"
+#include "util/string.h"
+#include "util/time.h"
+#include "util/trace.h"
+
+namespace {
+constexpr int kReportIdSize = 1;
+constexpr int kMaxHidErrorMessageSize = 512;
+} // namespace
+
+HidIoReport::HidIoReport(const unsigned char& reportId)
+        : m_reportId(reportId),
+          m_possiblyUnsendDataLatched(false) {
+}
+
+void HidIoReport::latchOutputReport(const QByteArray& data,
+        const mixxx::hid::DeviceInfo& deviceInfo,
+        const RuntimeLoggingCategory& logOutput) {
+    auto lock = lockMutex(&m_latchedOutputReportDataMutex);
+    if (m_possiblyUnsendDataLatched) {
+        qCDebug(logOutput) << "t:" << mixxx::Time::elapsed().formatMillisWithUnit()
+                           << " Skipped superseded OutputReport"
+                           << deviceInfo.formatName() << "serial #"
+                           << deviceInfo.serialNumberRaw() << "(Report ID"
+                           << m_reportId << ")";
+    }
+    m_latchedOutputReportData.replace(0, data.size(), data);
+    m_possiblyUnsendDataLatched = true;
+}
+
+bool HidIoReport::sendOutputReport(QMutex* pHidDeviceMutex,
+        hid_device* pHidDevice,
+        const mixxx::hid::DeviceInfo& deviceInfo,
+        const RuntimeLoggingCategory& logOutput) {
+    auto startOfHidWrite = mixxx::Time::elapsed();
+    QByteArray reportToSend;
+
+    auto lock = lockMutex(pHidDeviceMutex);
```

Comment:
Now the mutex is locked too early. It is also dangerous to lock different mutexes at the same time. If you don't follow a strict ordering you will run into deadlock situations. But we don't have this issue here. Both mutexes could be acquired/released one after a another.